src/main/java/com/ormanager/orm/OrmManager.java

```diff
@@ -347,4 +347,28 @@ private String getRecordId(T recordInDb) throws IllegalAccessException {
         }
         return "";
     }
+
+    public Object update(T o) throws IllegalAccessException {
+        if(getId(o)!=null && isRecordInDataBase(o)) {
+            LOGGER.info("This {} has been updated from Data Base.",
+                    o.getClass().getSimpleName());
+            return findById(getId(o), o.getClass()).get();
+        }
+        LOGGER.info("There is no such object with id in database or id of element is null.");
+        LOGGER.info("The object {} that was passed to the method was returned.",
+                o.getClass().getSimpleName());
+        return o;
+    }
+
+    private Serializable getId(T o) throws IllegalAccessException {
+
+        Optional<Field> optionalId = Arrays.stream(o.getClass().getDeclaredFields())
+                .filter(field -> field.isAnnotationPresent(Id.class))
+                .findAny();
+        if (optionalId.isPresent() && optionalId != null) {
```

Comment:
can optionalId be null at all? if yes optionalId.isPresent() will throw NPE and there is no need in second condition